### PR TITLE
Setup sqlite database

### PR DESCRIPTION
**Pull request by syntaxerron**
## Related Links:
- Issue links: 
https://www.notion.so/Setup-initial-DB-settings-c469f7fd9fdd4cbfa25a137ec7e92cfb

## Definition of Done
- [x] Setup sqlite database

## Notes:
- Delete the app installed
- Commands to run: 
     - `flutter pub get`
- To check database: 
     - Open the project in Android Studio
     - Run the application to open in an emulator
     - In Android Studio, go to View > Tool Windows > Device File Explorer from the Toolbar
     - Go to /data/data/com.example.sun_flutter_capstone/databases directory
     - Right click on file `expenses.db` and save the file
     - Open the file in a database manager or online (http://extendsclass.com/sqlite-browser.html#)

## Test View  Points
- [x] Setup sqlite database

### CODE REVIEW

**Review by AubreyV — APPROVED** [review]
## Approved
DB can be seen
Tables are complete
Schema is same with ERD
image

File: lib/models/model.dart [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    useSoftDeleting: false,
+    modelName: null,
+    fields: [
+      SqfEntityField('first_name', DbType.text, isNotNull: true),
+      SqfEntityField('last_name', DbType.text, isNotNull: true),
+      SqfEntityField('email', DbType.text, isNotNull: true),
+      SqfEntityField('currency', DbType.text, isNotNull: true, defaultValue: 'PHP'),
+      SqfEntityField('createdAt', DbType.datetime, isNotNull: true),
+      SqfEntityField('updatedAt', DbType.datetime, isNotNull: true),
+    ]);
```
Comment (inahsanjuan) [inline lib/models/model.dart]:
```suggestion
    ],);
```

File: lib/models/model.dart [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    primaryKeyName: 'id',
+    primaryKeyType: PrimaryKeyType.integer_auto_incremental,
+    useSoftDeleting: true,
+    modelName: null,
+    fields: [
+      SqfEntityField('name', DbType.text, isNotNull: true),
+      SqfEntityField('icon', DbType.text),
+      SqfEntityField('createdAt', DbType.datetime, isNotNull: true),
+      SqfEntityField('updatedAt', DbType.datetime, isNotNull: true),
+    ]);
```
Comment (inahsanjuan) [inline lib/models/model.dart]:
```suggestion
    ],);
```

File: lib/models/model.dart [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    primaryKeyName: 'id',
+    primaryKeyType: PrimaryKeyType.integer_auto_incremental,
+    useSoftDeleting: false,
+    fields: [
+      SqfEntityField('description', DbType.text, isNotNull: true),
+      SqfEntityField('amount', DbType.real, isNotNull: true, defaultValue: 0),
+      SqfEntityField('date', DbType.date, isNotNull: true),
+      SqfEntityField('createdAt', DbType.datetime, isNotNull: true),
+      SqfEntityField('updatedAt', DbType.datetime, isNotNull: true),
+    ]);
```
Comment (inahsanjuan) [inline lib/models/model.dart]:
```suggestion
    ],);
```

File: lib/models/model.dart [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    tableName: 'transactions',
+    primaryKeyName: 'id',
+    primaryKeyType: PrimaryKeyType.integer_auto_incremental,
+    useSoftDeleting: false,
+    fields: [
+      SqfEntityField('transaction_id', DbType.integer, isNotNull: true),
+      SqfEntityField('transaction_type', DbType.text, isNotNull: true),
+      SqfEntityField('createdAt', DbType.datetime, isNotNull: true),
+      SqfEntityField('updatedAt', DbType.datetime, isNotNull: true),
+    ]);
```
Comment (inahsanjuan) [inline lib/models/model.dart]:
```suggestion
    ],);
```

File: lib/models/model.dart [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+@SqfEntityBuilder(expenseDBModel)
+const expenseDBModel = SqfEntityModel(
+    modelName: 'ExpenseDBModel', // optional
+    databaseName: 'dbexpense.db',
+    databaseTables: [tableAccount, tableCategory, tableExpense, tableIncome, tableSpendingLimit, tableTransaction],
+    sequences: [],
+    bundledDatabasePath: null);
```
Comment (inahsanjuan) [inline lib/models/model.dart]:
```suggestion
    bundledDatabasePath: null,);
```

**Review by inahsanjuan — CHANGES_REQUESTED** [review]
@erronerron 
After updating code comments, please run 
```
flutter format . 
```

**Review by inahsanjuan — APPROVED** [review]
Comments: 2 (Compiled)
Review time: 20 minutes

**Pull request merged**